File: paper_artifacts/experiments/llr8/artifacts/sources/qwen38-fortran-skills/tsvc_2_vtvtv__submitted.f90
subroutine tsvc_2_vtvtv(a, b, c, len_1d, workspace, workspace_size) bind(C, name="tsvc_2_vtvtv_fp64")
  use iso_c_binding
  integer(c_int64_t), value, intent(in) :: len_1d, workspace_size
  real(c_double), intent(inout) :: a(len_1d)
  real(c_double), intent(in) :: b(len_1d)
  real(c_double), intent(in) :: c(len_1d)
  type(c_ptr), intent(in) :: workspace
  integer(c_int64_t) :: i

  if (len_1d <= 0) return

  !$omp parallel do simd
  do i = 1, len_1d
    a(i) = a(i) * b(i) * c(i)
  end do
end subroutine tsvc_2_vtvtv

File: paper_artifacts/experiments/llr8/artifacts/sources/qwen38-c-skills/tsvc_2_vtvtv__submitted.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <omp.h>

void tsvc_2_vtvtv_fp64(
    double *restrict a,
    const double *restrict b,
    const double *restrict c,
    const int64_t LEN_1D,
    uint8_t *restrict workspace,
    const int64_t workspace_size)
{
    (void)workspace;
    (void)workspace_size;
    if (LEN_1D < (1 << 18)) {
        for (int64_t i = 0; i < LEN_1D; ++i)
            a[i] = a[i] * b[i] * c[i];
    } else {
        #pragma omp parallel for simd schedule(static)
        for (int64_t i = 0; i < LEN_1D; ++i)
            a[i] = a[i] * b[i] * c[i];
    }
}
